jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/actions/DeployServerlessApplicationAction.kt

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.actions
+
+import com.intellij.openapi.actionSystem.AnAction
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.actionSystem.PlatformDataKeys
+import com.intellij.openapi.ui.Messages
+import icons.AwsIcons
+import software.aws.toolkits.resources.message
+
+class DeployServerlessApplicationAction : AnAction(
+        message("serverless.application.deploy"),
+        null,
+        AwsIcons.Resources.LAMBDA_FUNCTION) {
+    private var templateYamlRegex = Regex("template.y[a]?ml", RegexOption.IGNORE_CASE)
```

Comment:
Technically this matches `template$yaml`, regex also seems kinda wrong tool for the job